spec/lucky/url_helpers_spec.cr

```diff
@@ -0,0 +1,110 @@
+        view_for("/action").current_page?("/action/").should be_true
+        view_for("/action/").current_page?("/action").should be_true
+        view_for("/action/").current_page?("/action/").should be_true
+      end
+
+      it "tests if the path of a url matches request path or not" do
+        view_for("/")
+          .current_page?("https://example.com/")
+          .should be_true
+        view_for("/action")
+          .current_page?("https://example.com/action")
+          .should be_true
+        view_for("/action", host_with_port: "example.io")
+          .current_page?("https://example.com/action")
+          .should be_false
+        view_for("/action", host_with_port: "example.com:3000")
+          .current_page?("https://example.com/action")
+          .should be_false
+        view_for("/action", host_with_port: "example.com:3000")
+          .current_page?("https://example.com:3000/action")
+          .should be_true
+        view_for("/action", host_with_port: "example.com:3000")
+          .current_page?("http://example.com:3000/action")
+          .should be_true
+      end
+
+      it "only tests positive for get and head requests" do
+        view_for("/get", "GET").current_page?("/get").should be_true
+        view_for("/head", "HEAD").current_page?("/head").should be_true
+        view_for("/post", "POST").current_page?("/post").should be_false
+        view_for("/put", "PUT").current_page?("/put").should be_false
+        view_for("/patch", "PATCH").current_page?("/patch").should be_false
+        view_for("/delete", "DELETE").current_page?("/delete").should be_false
+      end
+
+      it "ignores query parameters by default" do
+        view_for("/action?order=desc&page=1").current_page?("/action")
+          .should be_true
+        view_for("/action").current_page?("/action?order=desc&page=1")
+          .should be_true
+        view_for("/action?order=desc&page=1").current_page?("/action/123")
+          .should be_false
+      end
+
+      it "checks query params if explicitly required" do
+        view_for("/action?order=desc&page=1")
+          .current_page?("/action?order=desc&page=1", check_query_params: true)
+          .should be_true
+        view_for("/action")
+          .current_page?("/action", check_query_params: true)
+          .should be_true
+        view_for("/action")
+          .current_page?("/action?order=desc&page=1", check_query_params: true)
+          .should be_false
+        view_for("/action?order=desc&page=1")
+          .current_page?("/action", check_query_params: true)
+          .should be_false
+      end
+    end
+
+    context "given a browser action" do
+      it "tests if the path matches or not" do
+        view_for("/pages/123").current_page?(Pages::Show.with(123))
+          .should be_true
+        view_for("/pages/123").current_page?(Pages::Show.with(12))
+          .should be_false
+        view_for("/pages").current_page?(Pages::Index)
+          .should be_true
```

Comment:
Thoughts on adding a spec where you include an anchor and another for an action with a query param?

Reply:
They are now added.